nucypher/policy/conditions/lingo.py

```diff
@@ -93,15 +93,15 @@ class ReturnValueTestSchema(CamelCaseSchema):
         def make(self, data, **kwargs):
             return ReturnValueTest(**data)
 
-    def __init__(self, comparator: str, value: Any, key: Optional[Union[int, str]] = None):
+    def __init__(self, comparator: str, value: Any, key: int = None):
         if comparator not in self.COMPARATORS:
             raise self.InvalidExpression(
                 f'"{comparator}" is not a permitted comparator.'
             )
 
-        if not isinstance(key, (int, str)) and key is not None:
+        if not isinstance(key, int) and key is not None:
```

Comment:
Minor nitpick, I would reverse the order to read as "is key set, and then is it an int" - but up to you.

```suggestion
        if key is not None and not isinstance(key, int):
```